pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1712,48 +1757,102 @@ END
 $$
 LANGUAGE PLPGSQL;
 
---Decompression should take place in a procedure because we don't want locks held across
---decompress_chunk calls since that function takes some heavier locks at the end.
+CALL execute_everywhere($ee$
+    --Decompression should take place in a procedure because we don't want locks held across
+    --decompress_chunk calls since that function takes some heavier locks at the end.
+    CREATE OR REPLACE PROCEDURE SCHEMA_CATALOG.do_decompress_chunks_after(metric_table NAME, min_time TIMESTAMPTZ)
+    AS $$
+    DECLARE
+        chunk_row record;
+        dimension_row record;
+        hypertable_row record;
+        min_time_internal bigint;
+    BEGIN
+        SELECT h.* INTO STRICT hypertable_row FROM _timescaledb_catalog.hypertable h
+        WHERE table_name = metric_table AND schema_name = 'SCHEMA_DATA';
+
+        SELECT d.* INTO STRICT dimension_row FROM _timescaledb_catalog.dimension d WHERE hypertable_id = hypertable_row.id ORDER BY id LIMIT 1;
+
+        SELECT _timescaledb_internal.time_to_internal(min_time) INTO STRICT min_time_internal;
+
+        FOR chunk_row IN
+            SELECT c.*
+            FROM _timescaledb_catalog.dimension_slice ds
+            INNER JOIN _timescaledb_catalog.chunk_constraint cc ON cc.dimension_slice_id = ds.id
+            INNER JOIN _timescaledb_catalog.chunk c ON cc.chunk_id = c.id
+            WHERE dimension_id = dimension_row.id
+            -- the range_starts are inclusive
+            AND min_time_internal >= ds.range_start
+            AND c.compressed_chunk_id IS NOT NULL
+            ORDER BY ds.range_start
+        LOOP
+
+            --lock the chunk exclusive.
+            EXECUTE format('LOCK %I.%I;', chunk_row.schema_name, chunk_row.table_name);
+            --double check it's still compressed.
+            PERFORM c.*
+            FROM _timescaledb_catalog.chunk c
+            WHERE c.id = chunk_row.id AND c.compressed_chunk_id IS NOT NULL;
+
+            IF FOUND THEN
+                RAISE NOTICE 'Promscale is decompressing chunk: %.%', chunk_row.schema_name, chunk_row.table_name;
+                PERFORM decompress_chunk(format('%I.%I', chunk_row.schema_name, chunk_row.table_name)::regclass);
+            END IF;
+
+            COMMIT;
+        END LOOP;
+    END;
+    $$ LANGUAGE PLPGSQL;
+$ee$);
+
 CREATE OR REPLACE PROCEDURE SCHEMA_CATALOG.decompress_chunks_after(metric_table NAME, min_time TIMESTAMPTZ)
+AS $proc$
+BEGIN
+    -- in early versions of timescale multinode the access node catalog does not
+    -- store whether chunks were compressed, so we need to run the actual search
+    -- for nodes in need of decompression on the data nodes
+    IF SCHEMA_CATALOG.is_multinode() THEN
+        CALL distributed_exec(
+            format(
+                $dist$ CALL do_decompress_chunks_after(%s, %s) $dist$,
+                quote_literal(metric_table), quote_literal(min_time)),
```

Comment:
```suggestion
                $dist$ CALL do_decompress_chunks_after(%L, %L) $dist$,
                metric_table, min_time),
```

Reply:
I had a specific reason I didn't want to do that, can I remember what it was...

Reply:
well let's change it then unless there is a reason